roles/datadog/tasks/pg_stats.yml

```diff
@@ -85,34 +111,52 @@
   when: integration_exists.stdout != '1'
 
 - name: check pg_stat_activity function
-  command: "psql {{ db }} -c \"SELECT * FROM pg_catalog.pg_proc WHERE proname = 'pg_stat_activity';\""
-  become: yes
+  command: "/usr/bin/psql \
+    -h {{ db_host }} \
+    -p {{ db_port }} \
+    -U {{ db_user }} \
+    -d {{ db }} -c \"SELECT * FROM pg_catalog.pg_proc WHERE proname = 'pg_stat_activity';\""
+  become: db_host == 'localhost'
+  environment:
+    PGPASSWORD: "{{ db_password }}"
   become_user: postgres
   register: pg_stat_activity_exist
   changed_when: False
 
 - name: add pg_stat_activity function
-  command: "psql {{ db }} -c \
+  command: "/usr/bin/psql  -h {{ db_host }} -p {{ db_port }} -U {{ db_user }} -d {{ db }} -c \
             'CREATE FUNCTION pg_stat_activity() RETURNS SETOF pg_catalog.pg_stat_activity AS $$
               SELECT * from pg_catalog.pg_stat_activity;
             $$ LANGUAGE sql VOLATILE SECURITY DEFINER;'"
-  become: yes
-  become_user: postgres
+  become: db_host == 'localhost'
   when: pg_stat_activity_exist.stdout.find('0 rows') != -1
+  environment:
+    PGPASSWORD: "{{ db_password }}"
   notify: restart postgres
 
 - name: check pg_stat_activity_dd view
-  command: "psql {{ db }} -c \"SELECT * FROM pg_catalog.pg_views WHERE viewname = 'pg_stat_activity_dd';\""
-  become: yes
+  command: "/usr/bin/psql -h {{ db_host }} \
+    -p {{ db_port }} \
+    -U {{ db_user }} \
+    -d {{ db }} \
+    -c \"SELECT * FROM pg_catalog.pg_views WHERE viewname = 'pg_stat_activity_dd';\""
+  become: db_host == 'localhost'
   become_user: postgres
   register: pg_stat_activity_dd_exist
+  environment:
+    PGPASSWORD: "{{ db_password }}"
   changed_when: False
 
 - name: add pg_stat_activity_dd view
```

Comment:
Has this playbook been re-tested on a server that _doesn't_ have a separate database to make sure it still works?

Reply:
:point_right: https://github.com/openfoodfoundation/ofn-install/pull/734#issuecomment-845863490

Reply:
Right, but this code only gets executed conditionally depending on what configurations the server has, and in the case of those 2 staging servers this specific part will be skipped. I have a suspicion it might not work now...

Reply:
Possibly. This is still WIP.